main_test.go

```diff
@@ -3,152 +3,209 @@ package main
+	actual, err := parseContent(nil, []string{})
 	if err != nil {
-		t.Errorf("Failed to read from clipboard: %v", err)
+		t.Fatalf("TestParseContent_Stdin: Expected no error, got %v", err)
 	}
 
-	if copied != content {
-		t.Errorf("Expected %s but got %s", content, copied)
+	// Verify the actual output matches the expected output
+	if actual != expected {
+		t.Errorf("TestParseContent_Stdin: Expected %s but got %s", expected, actual)
 	}
 }
 
-func TestParseContentDirectText(t *testing.T) {
-	directText := "Direct text input"
-	contentStr, err := parseContent(&directText, []string{})
+// TestParseContent_File verifies parseContent handles file input correctly.
+func TestParseContent_File(t *testing.T) {
+	// Content to write to the temporary file
+	content := "Content from file"
+	file := createTempFile(t, content)
+	defer os.Remove(file.Name()) // Clean up: remove the temporary file after the test
+
+	// Expected content after parsing the file
+	expected := "Content from file\n"
+
+	// Execute the function under test with the temporary file as input
+	actual, err := parseContent(nil, []string{file.Name()})
 	if err != nil {
-		t.Errorf("Expected no error, got %v", err)
+		t.Fatalf("TestParseContent_File: Expected no error, got %v", err)
 	}
-	if contentStr != directText {
-		t.Errorf("Expected contentStr to be '%s', got '%s'", directText, contentStr)
+
+	// Verify the actual output matches the expected output
+	if actual != expected {
+		t.Errorf("TestParseContent_File: Expected %s but got %s", expected, actual)
 	}
 }
 
-func TestParseContentStdin(t *testing.T) {
-	stdinInput := "Stdin input\n"
-	r, w, _ := os.Pipe()
-	originalStdin := os.Stdin
-	defer func() {
-		os.Stdin = originalStdin
-		r.Close()
-		w.Close()
-	}()
-	os.Stdin = r
-
-	// Write the input to the pipe
-	w.WriteString(stdinInput)
-	w.Close()
+// stringPtr creates a pointer to a string.
+func stringPtr(s string) *string {
+	return &s
+}
 
-	contentStr, err := parseContent(nil, []string{})
+// createTempFile creates a temporary file for testing purposes.
+func createTempFile(t *testing.T, content string) *os.File {
+	file, err := os.CreateTemp(t.TempDir(), "testfile")
 	if err != nil {
-		t.Errorf("Expected no error, got %v", err)
+		t.Fatalf("createTempFile: Failed to create temp file: %v", err)
 	}
-	if contentStr != stdinInput {
-		t.Errorf("Expected contentStr to be '%s', got '%s'", stdinInput, contentStr)
+	defer file.Close() // Ensure the file is closed
+
+	_, err = file.WriteString(content)
+	if err != nil {
+		t.Fatalf("createTempFile: Failed to write to temp file: %v", err)
 	}
-}
 
-func TestParseContentFromFile(t *testing.T) {
-	contentFromFile := "Content from file"
-	fakeFile := createTempFile(contentFromFile)
-	defer os.Remove(fakeFile.Name())
+	return file
+}
 
-	contentStr, err := parseContent(nil, []string{fakeFile.Name()})
+// replaceStdin replaces os.Stdin with a pipe for testing purposes.
+func replaceStdin(t *testing.T) (*os.File, *os.File, func()) {
```

Comment:
Missing t.Helper call here

```suggestion
func replaceStdin(t *testing.T) (*os.File, *os.File, func()) {
t.Helper ()
```